torch/distributed/_functional_collectives.py

```diff
@@ -600,6 +600,17 @@ def __tensor_unflatten__(inner_tensors, meta, outer_size, outer_stride):
         elem = inner_tensors["elem"]
         return AsyncCollectiveTensor(elem)
 
+    def __coerce_same_metadata_as_tangent__(
+        self, expected_metadata: Any, expected_type: Optional[Type] = None
+    ):
+        if expected_type is not torch.Tensor:
+            raise RuntimeError("Not implemented")
+
+        t = self.trigger_wait()
+        while isinstance(t, AsyncCollectiveTensor):
```

Comment:
nit: we shouldn't need this while loop either (I think we will be a in a very weird place if there are every nested AsyncCollectiveTensors, so it seems pointless to defensively program for it)